src/lua/digest.lua

```diff
@@ -267,4 +311,74 @@ m['aes256cbc'] = {
     end
 }
 
+local function check_unsigned(value)
+    local value_type = type(value)
+    if value_type == 'number' then
+        if value >= 0 and value < 2^53 and math.floor(value) == value then
+            return value
+        end
+    elseif value_type == 'cdata' and ffi.istype('uint64_t', value) then
+        return value
+    end
+end
+
+for _, var in ipairs({'32', '64'}) do
+    local xxHash
+
+    local update_fn_name = 'XXH' .. var .. '_update'
+    local digest_fn_name = 'XXH' .. var .. '_digest'
+    local reset_fn_name = 'XXH' .. var .. '_reset'
+    local copy_fn_name = 'XXH' .. var .. '_copyState'
+    local methods = {
+        update = function(self, str)
+            if type(str) ~= 'string' then
+                error("Usage xxhash" .. var .. ":update(string)")
+            end
+            return ffi.C[update_fn_name](self, str, #str) == ffi.C.XXH_OK
+        end,
+        result = function(self)
+            return ffi.C[digest_fn_name](self)
+        end,
+        clear = function(self, seed)
+            if check_unsigned(seed) == nil then
+                error("Usage xxhash" .. var .. ":clear(unsigned number)")
+            end
+            return ffi.C[reset_fn_name](self, seed) == ffi.C.XXH_OK
+        end,
+        copy = function(self)
+            local copy = xxHash.new()
+            ffi.C[copy_fn_name](copy, self)
+            return copy
+        end,
+    }
+
+    local XXH_state_t = ffi.typeof('XXH' .. var .. '_state_t')
+    ffi.metatype(XXH_state_t, {
+        __index = function(_, key)
+            return methods[key]
+        end,
+        __tostring = function(self)
+            return string.format('value: %s', self:result())
+        end,
+    })
+
+    xxHash = {
+        new = function()
+            return ffi.new(XXH_state_t)
+        end
```

Comment:
It looks counterintuitive that I need to call `clear()` right after `new()` just to set the seed. Better add it as an optional parameter for `new`, like pmur does.

Reply:
Agree, I'll fix it

Reply:
Currently seed is an optional value and it's value 0. Description is updated.